app/controllers/v0/facilities/ccp_controller.rb

```diff
@@ -1,10 +1,26 @@
 # frozen_string_literal: true
 
 class V0::Facilities::CcpController < FacilitiesController
+  before_action :validate_params,
+                :validate_no_services_without_type,
+                only: [:index]
+
   before_action :validate_id, only: [:show]
 
+  def index
+    case params[:type]
+    when 'cc_provider'
+      provider_locator
```

Comment:
I wonder if it would be clearer here to return the object and the pages and put the render method here in the controller action where you'd typically find it:

```
providers, pages = provider_locator
render json: providers,
	each_serializer: ProviderSerializer,
     meta: { pagination: pages }
```

or even

```
providers, = provider_locator
render json: providers,
	each_serializer: ProviderSerializer,
     meta: { pagination: paginate_providers(providers }
```

Reply:
This is an odd one. It's similar to the original implementation. This really shouldn't have pagination since we can't get page 2 from PPMS.